contracts/payloads/IGP52/PayloadIGP52.sol

```diff
@@ -0,0 +1,939 @@
+        TYPE_2,
+        TYPE_3,
+        TYPE_4
+    }
+
+    struct Vault {
+        address vault;
+        TYPE vaultType;
+        address supplyToken;
+        address borrowToken;
+        uint256 baseWithdrawalLimitInUSD;
+        uint256 baseBorrowLimitInUSD;
+        uint256 maxBorrowLimitInUSD;
+    }
+
+    constructor() {
+        ADDRESS_THIS = address(this);
+    }
+
+    function propose(string memory description) external {
+        require(
+            msg.sender == PROPOSER ||
+                msg.sender == TEAM_MULTISIG ||
+                address(this) == PROPOSER_AVO_MULTISIG ||
+                address(this) == PROPOSER_AVO_MULTISIG_2 ||
+                address(PROPOSER_AVO_MULTISIG_3) == PROPOSER_AVO_MULTISIG_3,
+            "msg.sender-not-allowed"
+        );
+
+        uint256 totalActions = 1;
+        address[] memory targets = new address[](totalActions);
+        uint256[] memory values = new uint256[](totalActions);
+        string[] memory signatures = new string[](totalActions);
+        bytes[] memory calldatas = new bytes[](totalActions);
+
+        // Action 1: call executePayload on timelock contract to execute payload related to Fluid.
+        targets[0] = address(TIMELOCK);
+        values[0] = 0;
+        signatures[0] = "executePayload(address,string,bytes)";
+        calldatas[0] = abi.encode(ADDRESS_THIS, "execute()", abi.encode());
+
+        uint256 proposedId = GOVERNOR.propose(
+            targets,
+            values,
+            signatures,
+            calldatas,
+            description
+        );
+
+        require(proposedId == PROPOSAL_ID, "PROPOSAL_IS_NOT_SAME");
+    }
+
+    function execute() external {
+        require(address(this) == address(TIMELOCK), "not-valid-caller");
+
+        // Action 1: Add 0.5M Rewards to fUSDC and fUSDT
+        action1();
+
+        // Action 2: Set ETH-weETH dex pool and vault initial limits.
+        action2();
+
+        // Action 3: Set USDe and sUSDe handler
+        action3();
+
+        // Action 4: Update GHO, USDC and USDT market rate curve on liquidity.
+        action4();
+    }
+
+    function verifyProposal() external view {}
+
+    /**
+     * |
+     * |     Proposal Payload Actions      |
+     * |__________________________________
+     */
+
+    /// @notice Action 1: Add 0.5M Rewards to fUSDC and fUSDT
+    function action1() internal {
+        address[] memory protocols = new address[](2);
+        address[] memory tokens = new address[](2);
+        uint256[] memory amounts = new uint256[](2);
+
+        address REWARDS_ADDRESS = address(0); // TODO: update
+
+        { /// fUSDC
+            IFTokenAdmin(F_USDC).updateRewards(REWARDS_ADDRESS);
+
+            uint256 allowance = IERC20(USDC_ADDRESS).allowance(
+                address(FLUID_RESERVE),
+                F_USDC
+            );
+
+            protocols[0] = F_USDC;
+            tokens[0] = USDC_ADDRESS;
+            amounts[0] = allowance + (500_000 * 1e6);
```

Comment:
increasing by 200k will be enough here because we are replacing the 300k ongoing for which we already increased. same for USDT